### PR TITLE
[PluggableDevice] Fix tf.config.set_visible_device for PluggableDevice

### PR DESCRIPTION
**Pull request by jzhoulon**
tf.config.set_visible_device() is not work for PluggableDevice but can work normally through tf.ConfigProto()
work example:
```
config = tf.ConfigProto()
config.gpu_options.visible_device_list = "1"
with tf.Session(config) as sess:
```
not work example:
```
devs = tf.config.list_physical_devices('XPU')
tf.config.set_visible_devices(devs[1], 'XPU')
```
 The root cause is currently python implementation in v2 API will filter out non-gpu devices in GPUOptions, this PR will enable it if the device is GPU or PluggableDevice.

### CODE REVIEW

**Comment by jzhoulon** [comment]
@penpornk we find that tf.config.set_visible_device is not working for pluggable device when we do multi-cards training with horovod. We find that it was due to some 'GPU' hardcode in /tensorflow/eager/context.py, we changed the device check as 'GPU' + pluggable_device, can you help to review it? Thanks.

By the way, I saw there was a pylint error but not related with this PR. do I need to take care of it? Thanks
```
pylint error
tensorflow/python/eager/context.py:81:0: C0301: Line too long (83/80) (line-too-long)
```
```
code:
  79 # This flag and the associated environment var are transient and will eventually
  80 # be removed, once this experiment is enabled by default.
  81 _RUN_EAGER_OP_AS_FUNCTION_ENABLED = os.getenv("TF_RUN_EAGER_OP_AS_FUNCTION", False)
```

**Comment by Jianhui-Li** [comment]
@penpornk

File: tensorflow/python/eager/context.py [hunk 1]
```diff
@@ -1204,7 +1205,13 @@ def _compute_gpu_options(self):
     virtual_devices = []
     gpu_index = -1
     memory_growths = set()
-    for dev in self.list_physical_devices("GPU"):
+    gpu_devices = self.list_physical_devices("GPU")
+    pluggable_devices = self._pluggable_devices
+    compatible_devices = gpu_devices
+    for dev in pluggable_devices:
+      if dev not in gpu_devices:
```
Comment (penpornk) [inline tensorflow/python/eager/context.py:1212]: Could you please give an example of how a device can be in both `gpu_devices` and `pluggable_devices` lists? IIUC, the lists should be disjoint by construction.

File: tensorflow/python/eager/context.py [hunk 2]
```diff
@@ -1430,10 +1437,16 @@ def _initialize_physical_devices(self, reinitialize=False):
       self._physical_device_to_index = {
           p: i for i, p in enumerate(self._physical_devices)
       }
+      pluggable_devs = pywrap_tfe.TF_ListPluggablePhysicalDevices()
```
Comment (penpornk) [inline tensorflow/python/eager/context.py:1440]: Please add a comment that we maintain a separate list just so we can check which device is a PluggableDevice in `set_memory_growth`.

File: tensorflow/python/eager/context.py [hunk 3]
```diff
@@ -1430,10 +1437,16 @@ def _initialize_physical_devices(self, reinitialize=False):
       self._physical_device_to_index = {
```
Comment (penpornk) [inline tensorflow/python/eager/context.py:1437]: Will you need this index map too? If so, maybe we should add pluggable devices directly to `_physical_devices`.

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR!

File: tensorflow/python/eager/context.py [hunk 4]
```diff
@@ -1204,7 +1205,13 @@ def _compute_gpu_options(self):
     virtual_devices = []
     gpu_index = -1
     memory_growths = set()
-    for dev in self.list_physical_devices("GPU"):
+    gpu_devices = self.list_physical_devices("GPU")
+    pluggable_devices = self._pluggable_devices
+    compatible_devices = gpu_devices
+    for dev in pluggable_devices:
+      if dev not in gpu_devices:
+        compatible_devices.append(dev)
```
[the remark this answers is not in the document]
Comment (jzhoulon) [reply tensorflow/python/eager/context.py:1213]: Thanks for the comments, we once have tried with set, however, set will make the order of device become "unordered", which will make the filtered device incorrectly, for example, [PluggableDevice0, PluggableDevice1] may become [PluggableDevice1, PluggableDevice0], when set_visibile_device（“PluggableDevice:1”）, the physical device may be PluggableDevice0

File: tensorflow/python/eager/context.py (hunk 1 above)
Comment (jzhoulon) [reply tensorflow/python/eager/context.py:1212]: When PluggableDevice is registered with "GPU" device type(although currently is not supported), the "GPU" would be in gpu_devices as well as pluggable device list.

File: tensorflow/python/eager/context.py (hunk 3 above)
Comment (jzhoulon) [reply tensorflow/python/eager/context.py:1437]: _physical_devices already contains the pluggable devices, the pluggable device list here is to justify whether the device got from _physical_devices is a PluggableDevice.
for example:
```
1639     if dev.device_type != "GPU" and dev not in self._pluggable_devices:
1640       raise ValueError("Cannot set memory growth on non-GPU and non-Pluggable devices")

```

File: tensorflow/python/eager/context.py (hunk 1 above)
Comment (penpornk) [reply tensorflow/python/eager/context.py:1213]: Makes sense. Thank you!

File: tensorflow/python/eager/context.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (penpornk) [reply tensorflow/python/eager/context.py:1214]: I see. Maybe we should add a comment about this in the code later too.

**Review by penpornk — APPROVED** [review]

**Pull request merged**